provider/provider.go

```diff
@@ -18,13 +18,20 @@ package provider
 
 import (
 	"context"
+	"errors"
 	"net"
 	"strings"
 
 	"sigs.k8s.io/external-dns/endpoint"
 	"sigs.k8s.io/external-dns/plan"
 )
 
+var SoftError error = errors.New("soft error")
+
+func WrapSoftError(err error) error {
```

Comment:
In Go you can not have func and var with the same name.

Reply:
Yes, you're right.
Wdyt of renaming var from `SoftError` to `WrapSoftError` and func from `WrapSoftError` to `SoftError`, so it can be called with `provider.SoftError(...)` ?

Reply:
That sounds more weird to me. Maybe `NewSoftError(err error)` ?

Reply:
done